### PR TITLE
Add [[nodiscard]] attribute to all const member functions

### PR DESCRIPTION
**Pull request by lava**
This commit was generated automatically by running

    find libvast/src/ -name '*.cpp' -exec clang-tidy -p . --header-filter=/vast/ --checks='-*,modernize-use-nodiscard' --fix {} \;

followed by

    git format-branch

against the source tree.

###  :memo: Checklist

- [ ] All user-facing changes have changelog entries.
- [ ] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [ ] The PR description contains instructions for the reviewer, if necessary.

### CODE REVIEW

**Comment by tobim** [comment]
Can we please hold off on this until #1514 is merged?

**Review by dominiklohmann — APPROVED** [review]
Per discussion in Slack, this is exempt from the feature freeze period as it is a non-functional, tool-created refactoring that is 100% covered by our CI via `-Werror`.

**Pull request merged**
